crates/runner-nordic/src/main.rs

```diff
@@ -159,6 +200,80 @@ fn main() -> ! {
     Scheduler::<Board>::run();
 }
 
+pub struct RadioMetadata {
+    ticks: u32,
+    freq: u16,
+    rssi: i8,
+    pdu_type: u8,
+}
+
+impl RadioMetadata {
+    pub fn len(&self) -> usize {
```

Comment:
If I understand correctly, those `len()` functions are only used for debugging right? If yes, then let's have them `#[cfg(feature = "debug")]`. Also, not sure they need to be `pub` given they're at the root.